### PR TITLE
Update dependencies and fix rubocop

### PR DESCRIPTION
**Pull request by delphaber**
Fixed some rubocop issues to make CI green again

### CODE REVIEW

**Comment by delphaber** [comment]
It is still failing because rubocop does not support ruby 2.4.

I see that in the readme the support for ruby 2.4 has been deprecated. Shall I remove all reference to ruby 2.4 from the repo as well? Also, shall I remove the tests about Rails 4?

**Comment by brunofacca** [comment]
Hi Fabrizio. Thank you for your contribution.
Yes, please remove support for Ruby 2.4 and Rails 4. 
If you're up to it, we could even update the CI file to only support:
* Ruby 2.7, 3.0, 3.1, 3.2, and head
* Rails 6.0, 6.1, 7.0, and head

What do you think? Other people's opinions are welcome too.
Don't worry if you don't have the bandwidth to do it, I can (eventually) do it.

**Comment by delphaber** [comment]
It makes sense :) I can do it!

**Comment by brunofacca** [comment]
Awesome, thanks! :)

**Comment by delphaber** [comment]
All green! But I don't think travis is using the real ruby head version. If you open a test matrix in travis you see this

`$ ruby --version

ruby 3.0.0dev (2020-12-20T06:13:25Z master b911509a08) [x86_64-linux]`

Any idea? 🤔

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -8,11 +8,18 @@
 require 'active_record'
 require 'active_record_query_trace'
 require 'db/setup'
-require 'pry'
-require 'pry-byebug'
+require 'debug'
```
Comment (delphaber) [inline spec/spec_helper.rb:11]: debug is the new pry-byebug :D

**Comment by delphaber** [comment]
Oh I still need to update the Readme. Maybe I'll do it after the review of the PR :)

**Comment by brunofacca** [comment]
Thank you very much :). 
I'll review within the next couple of days.

File: Gemfile [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+
+gem 'rake', '~> 13.0'
+
+gem 'debug', '~> 1.8'
+gem 'rspec', '~> 3.12'
+gem 'rubocop', '~> 1.55'
+gem 'rubocop-performance', '~> 1.18'
+gem 'rubocop-rails', '~> 2.20'
+gem 'rubocop-rspec', '~> 2.22'
+gem 'simplecov', '>= 0.22.0'
```
Comment (brunofacca) [inline Gemfile:19]: Are those necessary? Can they be pulled from the gemspec so that we want to update the gem versions, we don't need to do that in two places?

File: active_record_query_trace.gemspec [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+  spec.metadata['homepage_uri'] = spec.homepage
+
+  # Specify which files should be added to the gem when it is released.
+  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
+  spec.files = Dir.chdir(__dir__) do
+    `git ls-files -z`.split("\x0").reject do |f|
+      (File.expand_path(f) == __FILE__) ||
+        f.start_with?(*%w[bin/ test/ spec/ features/ .git .travis.yml appveyor Gemfile])
+    end
+  end
```
Comment (brunofacca) [inline active_record_query_trace.gemspec:27]: Why is this necessary?

**Review by brunofacca — COMMENTED** [review]
Great job! The PR looks good to me. I just left a couple of questions. Other than that, should be ready to merge once the readme is updated.

**Comment by brunofacca** [comment]
Great job! The PR looks good to me. I just left a question regarding the repetition of the gem names and versions across multiple `Gemfile`s. Other than that, should be ready to merge once the readme is updated.

> All green! But I don't think travis is using the real ruby head version. If you open a test matrix in travis you see this
> 
> `$ ruby --version
> 
> ruby 3.0.0dev (2020-12-20T06:13:25Z master b911509a08) [x86_64-linux]`
> 
> Any idea? 🤔

You're right, `ruby-head` should not be pointing to `3.0.0dev`. I took a quick look and was unable to find out why, will look again when I have more time. But we can merge the PR regardless of this issue. Having Rubocop pass and the latest Ruby and Rails versions in the CI is already a big win.

File: active_record_query_trace.gemspec (hunk 3 above)
Comment (delphaber) [reply active_record_query_trace.gemspec:27]: Not a specific reason. I just copy-pasted a a gemspec from a new `bundle gem foobar` installation. I can switch batch to `spec.files = Dir['lib/**/*']` if you prefer!

File: Gemfile (hunk 2 above)
Comment (delphaber) [reply Gemfile]: done!

File: active_record_query_trace.gemspec (hunk 3 above)
Comment (delphaber) [reply active_record_query_trace.gemspec]: Done!

**Comment by brunofacca** [comment]
So satisfying to see the green build again haha
Great work! Thanks again!
